### PR TITLE
ECO-1022: Redirect to Sofort and Paydirect page after placing an order

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
- Developer(s): @aleksey-kotsuba

- Peer Reviewer:

- Ticket: https://spryker.atlassian.net/browse/ECO-1022

- Project PR: https://github.com/spryker-eco/computop

- Academy PR:

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Computop              | patch (new)           |                       |

#### Release Notes

-----------------------------------------

#### Module Computop

_**Patch:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated.
- [ ] The change is isolated and not mixed with any cleanups or other changes.
- [ ] New code fits to the architecture rules.
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer).

### Improvements
* Transaction ID has been limited to 20 chars for paydirect payment method.

### CODE REVIEW

File: src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/PaydirektMapper.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ protected function createTransferWithUnencryptedValues(TransferInterface $quoteT
         $computopPaymentTransfer = new ComputopPaydirektPaymentTransfer();
         
         $computopPaymentTransfer->setCapture(ComputopConfig::CAPTURE_MANUAL_TYPE);
-        $computopPaymentTransfer->setTransId($this->generateTransId($quoteTransfer));
+        $computopPaymentTransfer->setTransId(substr($this->generateTransId($quoteTransfer), 0, 20));
```
Comment (profuel) [inline src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/PaydirektMapper.php]: Please Move this logic of 20 symbols to `generateTransId`  function.
If you think it's likely to change in future, create separate function, still calling it inside `generateTransId` .
Also, move this 20 to module constants (not config in `config_default`).

File: src/SprykerEco/Yves/Computop/Mapper/Init/PrePlace/CreditCardMapper.php [hunk 2]
```diff
@@ -83,8 +83,8 @@ protected function getUrlToComputop($merchantId, $data, $length)
             ComputopApiConfig::DATA => $data,
             ComputopApiConfig::LENGTH => $length,
             ComputopApiConfig::URL_BACK => $this->getAbsoluteUrl(
-                    $this->application->path($this->config->getCallbackFailureRedirectPath())
-                )
+                $this->application->path($this->config->getCallbackFailureRedirectPath())
```
Comment (profuel) [inline src/SprykerEco/Yves/Computop/Mapper/Init/PrePlace/CreditCardMapper.php]: Minor: You could fix this once and simply merge into `dev` dranch

**Comment by profuel** [comment]
I don't see changes related to 
```We can remove "OrderDesc=Test:0000" parameter on "Sofort" payment method. Than there will be a redirect. But adding "OrderDesc=Test:0000" parameter was suggested by computop support. Without it other api calls for this payment method (f.e. refund) does not work properly.```

**Comment by aleksey-kotsuba** [comment]
Redirects were already implemented. I just have confirmed that, and fix problem with Paydirect payment method.

**Pull request merged**